File: driveBack/src/main/java/com/example/driveBack/dto/VehiclePreview.java
package com.example.driveBack.dto;

import com.example.driveBack.model.Driver;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class VehiclePreview {
    private Long id;
    private String brand;
    private Driver driver;
    private double distance;
    private double totalPrice;
    private double totalDistance;
    private double startPrice;
    private double pricePerKm;

    public VehiclePreview(VehiclePreviewI i){
        id = i.getId();
        brand = i.getBrand();
        driver = new Driver(i.getFirstName(), i.getLastName());
        distance = i.getDistance();
        startPrice = i.getStartPrice();
        pricePerKm = i.getPricePerKm();
    }

    public void setTotalPrice(){
        totalPrice = startPrice + totalDistance*pricePerKm;
    }


}

File: driveFront/src/app/model/vehicle-preview.ts
export interface VehiclePreviewI {
  id: number;
  brand: string;
  driver: {
    firstName: string;
    lastName: string;
  };
  distance: number;
  totalPrice: number;
  totalDistance: number;
  startPrice: number;
  pricePerKm: number;
}

export class VehiclePreview implements VehiclePreviewI {
  id: number = 0;
  brand: string = '';
  driver = { firstName: '', lastName: '' };
  distance: number = 0;
  totalPrice: number = 0;
  totalDistance: number = 0;
  startPrice: number = 0;
  pricePerKm: number = 0;
}
